mangaki/mangaki/admin.py

```diff
@@ -32,6 +56,63 @@ def make_nsfw(self, request, queryset):
         self.message_user(request, "%s désormais NSFW." % message_bit)
     make_nsfw.short_description = "Rendre NSFW les œuvres sélectionnées"
 
+    #update_tags_via_anidb
+    def anidb(self, request, queryset):
+        
+        if request.POST.get("post"):
+            
+            chosen_ids = request.POST.getlist('checks')
+            for anime_id in chosen_ids:
+                anime = Work.objects.get(id=anime_id)
+                retrieve_tags = anime.retrieve_tags()
+                deleted_tags = retrieve_tags["deleted_tags"]
+                added_tags = retrieve_tags["added_tags"]
+                updated_tags = retrieve_tags["updated_tags"]
+
+                anime = Work.objects.get(id=anime_id)
+                for  title, weight in added_tags.items():
+                    current_tag = Tag.objects.update_or_create(title=title)[0]
+                    TaggedWork(tag=current_tag, work=anime, weight=weight).save()
+
+                for title, weight in updated_tags.items(): 
+                    current_tag = Tag.objects.filter(title=title)[0]
+                    tag_work = TaggedWork.objects.get(tag=current_tag, work=anime, weight=weight[0])
+                    tag_work.delete()
+                    TaggedWork(tag=current_tag, work=anime, weight=weight[1]).save()
+    
+                for title, weight in deleted_tags.items():
+                    current_tag = Tag.objects.get(title=title)
+                    TaggedWork.objects.get(tag=current_tag, work=anime, weight=weight).delete()
+            self.message_user(request, "Modifications sur les tags faites")
+            return None
+        
+        for anime in queryset :
```

Comment:
Nit: ajouter ici un `select_related("category")` pour éviter une requête à chaque itération de la boucle.